### ISSUE TITLE
Cannot create a Network of more than one output

### ISSUE THREAD

**Issue by kelbazz** [mathletedev/rust-ml#1]
I cannot create a neural network with more than one output number for some reason. Here is the issue at the compilation:
`thread 'main' panicked at 'Invalid inputs length', src\network.rs:52:13`
And here is my code:
```rust
use activations::SIGMOID;
use network::Network;

pub mod activations;
pub mod matrix;
pub mod network;

fn main() {
	let inputs = vec![
		vec![1.0, 0.0, 0.0],
		vec![0.0, 1.0, 0.0]
	];
	let targets = vec![
		vec![0.0, 1.0, 0.0],
		vec![0.0, 0.0, 1.0]
	];
	
	let mut network = Network::new(vec![3, 4, 3], 0.5, SIGMOID);
	
	network.train(inputs, targets, 1000);

	println!("{:?}", network.feed_forward(vec![0.1, 0.0, 0.0]));
	println!("{:?}", network.feed_forward(vec![0.0, 1.0, 0.0]));
	println!("{:?}", network.feed_forward(vec![0.0, 0.0, 1.0]));
}
```

**Comment by neowsl** [comment]
Whoops... I forgot to test multiple outputs. It's fixed as of https://github.com/mathletedev/rust-ml/commit/66da8cefeadb5e863b445eb649871efb10037a78

_If you want an explanation_
Turns out, when doing the feed-forward, I converted the resultant matrix into a vector (effectively reducing a 2d array to a 1d array)
However, I did that by only taking the first row of the matrix. The matrix is a "column", not a "row":
```
[
  [a],
  [b],      vs.      [ [a, b, c] ]
  [c]
]
```
so taking the first row basically deleted all the info from the rest of the rows (meaning it works perfectly for one output, but doesn't work for multiple outputs)
The fix was simple: transpose the matrix before taking the first row, so the "column" matrix becomes a "row" matrix, and taking just the first row will preserve all the data

Thanks for pointing out this issue! Hope it works for you now!
